Source/SourceKittenFramework/String+SourceKitten.swift

```diff
@@ -80,11 +80,17 @@ extension NSString {
     */
     public func byteRangeToNSRange(start start: Int, length: Int) -> NSRange? {
         let string = self as String
-        return string.indexOfByteOffset(start).flatMap { stringStart in
-            return string.indexOfByteOffset(start + length).map { stringEnd in
-                return NSRange(location: stringStart, length: stringEnd - stringStart)
-            }
+        let startUTF8Index = string.utf8.startIndex.advancedBy(start)
+        let endUTF8Index = startUTF8Index.advancedBy(length)
+        
+        guard let startUTF16Index = startUTF8Index.samePositionIn(string.utf16),
```

Comment:
that's not what I mean. I mean are `utf16` views on Strings cached by the Swift standard library?

e.g. is it more efficient to do this:

```swift
let utf16View = string.utf16
guard let startUTF16Index = startUTF8Index.samePositionIn(utf16View),
      let endUTF16Index = endUTF8Index.samePositionIn(utf16View) else {
        return nil
}
```

or is that equivalent to this (current code):


```swift
guard let startUTF16Index = startUTF8Index.samePositionIn(string.utf16),
      let endUTF16Index = endUTF8Index.samePositionIn(string.utf16) else {
        return nil
}
```